### ISSUE TITLE
Deprecation - convert VersionUtils to long versionCode from int API

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#5018]
De-linting the codebase, this appears easy but because the versioncode is stored strongly-typed in preferences it affects forwards- and backwards-compatibility and so needs an accompanying test in order to make sure it doesn't cause problems.

**Comment by milyukhin** [comment]
Hi @mikehardy, can you please help me with the PR..
I've made changes based on the Compat framework, but I'm not quite sure what exactly you mean in your comments:

> // FIXME to use new API and change from int to long is very problematic. It's strongly typed in the XML and needs handling
// FIXME or it isn't backwards compatible - blows up development and may hurt users

specifically the part about strongly typed in XML, can you please clarify on this?

**Comment by mikehardy** [comment]
Hey! yeah - check out this result from direct inspection of the shared preferences XML in an AnkiDroid install on an emulator:

```
generic_x86:/data/data/com.ichi2.anki/shared_prefs # more com.ichi2.anki_preferences.xml                                                                         
<?xml version='1.0' encoding='utf-8' standalone='yes' ?>
<map>
    <boolean name="acra.legacyAlreadyConvertedTo4.8.0" value="true" />
    <string name="lastVersion">2.9beta2</string>
    <boolean name="acra.legacyAlreadyConvertedToJson" value="true" />
    <boolean name="acra.disable" value="true" />
    <string name="reportErrorMode">1</string>
    <int name="acra.lastVersionNr" value="20900202" />
</map>
```

You can see specifically the shared preferences XML "object" has an "int" type, and that's the problem. However we handle this needs to change actual data type and deal with the transition somehow

**Comment by milyukhin** [comment]
Aah, yeah, the shared preferences xml file... I've checked how the property `lastUpgradeVersion` (is this the only one we care about atm, right?) type changes with my PR (from string and int to long)

**Comment by mikehardy** [comment]
Fixed by #5387
